assets/src/components/PostSelector/PostSelector.js

```diff
@@ -1,103 +1,104 @@
-import { Component } from '@wordpress/element';
-import { compose } from '@wordpress/compose';
-const { apiFetch } = wp;
-const { addQueryArgs } = wp.url;
-import {
-  FormTokenField,
-} from '@wordpress/components';
+import { FormTokenField } from '@wordpress/components';
+import { dispatch, useSelect } from '@wordpress/data';
+const { __ } = wp.i18n;
 
-class PostSelector extends Component {
-  constructor(props) {
-    super(props);
+// Allows to query a custom endpoint with select('core') tools
+dispatch('core').addEntities( [{
+    baseURL: '/planet4/v1/all-published-posts',
+    kind: 'planet4/v1',
+    name: 'all-published-posts',
+    label: 'All published posts',
+}] );
 
-    this.handleChange = this.handleChange.bind(this);
-    this.getValue = this.getValue.bind(this);
-    this.state = {
-      posts: [],
-    };
-  }
+/**
+ * Post selector with autosuggestion
+ * Based on post type
+ */
+export const PostSelector = (attributes) => {
+  const {
+    label,
+    selected,
+    placeholder,
+    postType,
+    onChange,
+    maxLength,
+    maxSuggestions,
+  } = attributes;
 
-  componentDidMount() {
-    let args = {};
-    // If WPML is active then this variable contains the post language.
-    if (typeof icl_this_lang !== 'undefined') {
-      args = { post_language: icl_this_lang };
+  /**
+   * Fetch relevant posts for autosuggestions
+   */
+  const args = { per_page: -1, orderby: 'title', post_status: 'publish' };
+  const act_args = {
+    post_parent: window.p4ge_vars.planet4_options.act_page,
```

Comment:
Can this be `null` or `undefined`? What would happen then? 🤔

Reply:
Wasn't sure, but it could return all pages, so I changed the code to not run this specific request if there's no Act parent page